jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/BaseEnricher.java

```diff
@@ -55,6 +55,7 @@
     public static final String CREATE_EXTERNAL_URLS = "jkube.createExternalUrls";
     public static final String JKUBE_DOMAIN = "jkube.domain";
     public static final String JKUBE_ENFORCED_REPLICAS = "jkube.replicas";
+    public static final String DEFAULT_JOB_RESTART_POLICY = "OnFailure";
```

Comment:
Was this the default behavior when it was an optional field? do you have any reference?

Reply:
As pointed out in PR description, the default value is `Always` (if not specified) for most controllers. However, for `Job` it's mandatory to provide restart policy

https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy

Reply:
But what was the behavior for Job controllers when this value wasn't required. That's the question

Reply:
Right now, we're assuming it is `OnFailure`.

Reply:
It wasn't provided and Job was created without any `restartPolicy`. This used to get rejected by Kubernetes api server on apply phase

Reply:
I understand the issue. What I mean, is that I assumed that at some point this was working (however I'm doubting this more and more), so the kube API changed the requirements for Job Pod Templates at some point.

Reply:
OK, so my worries are confirmed, this feature never worked...

https://github.com/kubernetes/kubernetes/issues/41194